tests/test_evaluate_api.py

```diff
@@ -70,6 +73,209 @@ def _run_test(test):
+        self.app.post(
+            f"/v1/resource/{resource_type}/policy/read",
+            data=json.dumps({'policy': create_test_ResourcePolicy(
+                'read',
+                actions=['rt:read'],
+                resource_type=resource_type,
+                effect='Allow'
+            )}),
+            headers=admin_headers
+        )
+
+        # create a resource policy rw
+        self.app.post(
+            f"/v1/resource/{resource_type}/policy/rw",
+            data=json.dumps({
+                'policy': create_test_ResourcePolicy(
+                    'rw',
+                    actions=['rt:read', 'rt:write', 'rt:delete'],
+                    resource_type=resource_type,
+                    effect='Allow'
+                )},
+            ),
+            headers=admin_headers
+        )
+
+        # create a resource id
+        resp = self.app.post(
+            f'/v1/resource/{resource_type}/id/{resource_id}',
+            headers=admin_headers)
+        self.assertEqual(resp.status_code, 201)
+
+        # create a role with read
+        resp = self.app.post(
+            f'/v1/role',
+            data=json.dumps(
+                {'role_id': 'read',
+                 'policy': create_test_IAMPolicy(
+                     'read',
+                     actions=['rt:read'],
+                     resource_type=resource_type,
+                     effect='Allow'
+                 )}),
+            headers=admin_headers
+        )
+        self.assertEqual(resp.status_code, 201)
+
+        # create a role with rw
+        resp = self.app.post(
+            f'/v1/role',
+            data=json.dumps(
+                {'role_id': 'rw',
+                 'policy': create_test_IAMPolicy(
+                     'rw',
+                     actions=['rt:read', 'rt:write', 'rt:delete'],
+                     resource_type=resource_type,
+                     effect='Allow'
+                 )}),
+            headers=admin_headers
+        )
+        self.assertEqual(resp.status_code, 201)
+
+        # create a user
+        resp = self.app.post(
+            f'/v1/user',
+            data=json.dumps({'user_id': user}),
+            headers=admin_headers)
+        self.assertEqual(resp.status_code, 201)
+
+        # user fails to read resource
+        resp = self.app.post(
+            '/v1/policies/evaluate',
+            headers=admin_headers,
+            data=json.dumps(
+                {'principal': user,
+                 'action': ['rt:read'],
+                 'resource': [resource_arn]}
+            )
+        )
+        self.assertFalse(json.loads(resp.body)['result'])
+
+        # give the user role read
+        resp = self.app.put(
+            f'/v1/user/{user}/roles?action=add',
+            headers=admin_headers,
+            data=json.dumps(
+                {'roles': ['read']}
+            )
+
+        )
+        self.assertEqual(resp.status_code, 200)
+
+        # user fails to read resource
```

Comment:
Since it is a resource that we have applied resource bases access control, the user must have access to perform the read operation and the resource must allow the user to perform a read operation.

Reply:
In this case the user has a role that grants read operations, but he does not have access to the resource he want to read.